lib/galaxy/security/validate_user_input.py

```diff
@@ -75,11 +84,23 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
         domain = extract_domain(email, base_only=True)
         if domain in trans.app.config.email_domain_blocklist_content:
             message = "Please enter your permanent email address."
+    if not message and validate_domain:
+        domain = email.split('@', 1)[1]
```

Comment:
```suggestion
        domain = extract_domain(email)
```
Death to duplicate code?

Reply:
Oh yeah sure enough. I looked at `extract_domain()` thinking "well that chops off subdomains, which I don't want," totally ignoring the `base_only` param.